### PR TITLE
Rental Check Out Tests and Action

### PR DESCRIPTION
**Pull request by dnguye2**

### CODE REVIEW

File: app/controllers/rentals_controller.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+class RentalsController < ApplicationController
+  def checkout
+    # create new rental
+    rental = Rental.new(rental_params)
+
+    customer = Customer.find_by(id: params[:customer_id])
+    video = Video.find_by(id: params[:video_id])
+    
+    if !customer
```
Comment (theomoondev) [inline app/controllers/rentals_controller.rb]: I think we can do `if !customer || !video` here to dry this up a bit!

File: app/controllers/rentals_controller.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    end
+        
+    if video.available_inventory == 0
+      render json: {
+          errors: ['Bad Request']
+        }, status: :bad_request
+      return
+    end
+
+    Rental.check_out(rental.customer_id, rental.video_id)
```
Comment (theomoondev) [inline app/controllers/rentals_controller.rb:23]: I'm not sure if this would work, since rental hasn't been created yet, but could this be an instance method instead of a class method? Then we wouldn't need to pass in parameters because the method would know that customer_id refers to rental.customer_id, etc.

```suggestion
    rental.check_out
```

File: app/controllers/rentals_controller.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        }, status: :bad_request
+      return
+    end
+  end
+
+
+  private
+
+  def rental_params
+    return params.permit(:customer_id, :video_id, :due_date)
```
Comment (theomoondev) [inline app/controllers/rentals_controller.rb]: Do we need to permit :due_date since we're not getting it from the params? I guess it's a question of whether we want the user to be able to set a custom due date?

File: app/models/rental.rb [hunk 4]
```diff
@@ -3,4 +3,10 @@ class Rental < ApplicationRecord
   
   belongs_to :customer
   belongs_to :video
+
+  # checkout method
+  def self.check_out(customer_id, video_id)
```
Comment (theomoondev) [inline app/models/rental.rb:8]: Re: my comment above about this being a class method vs. instance method. Not sure if an instance method would work though, so maybe this is a bogus suggestion?

```suggestion
  def check_out
```

File: test/fixtures/videos.yml [hunk 5]
```diff
@@ -15,4 +15,10 @@ grand_budapest:
   overview: The Grand Budapest Hotel is a 2014 comedy-drama film written and directed by Wes Anderson, which explores tragedy, war, fascism, nostalgia, friendship, and loyalty. Ralph Fiennes leads a seventeen-actor ensemble cast as Monsieur Gustave H., famed concierge of a mountainside resort in the Republic of Zubrowka.
   release_date: 2014-03-28
   total_inventory: 9
-  available_inventory: 9
\ No newline at end of file
+  available_inventory: 9
+rested:
+  title: Restful in Seattle
+  overview: When interview week is over and C13 is probably going to get lots of rest
```
Comment (theomoondev) [inline test/fixtures/videos.yml:21]: HAHAHA I love this!! 🥰🌻

File: app/controllers/rentals_controller.rb (hunk 1 above)
Comment (dnguye2) [reply app/controllers/rentals_controller.rb]: Good point! I'll go ahead and change it and push it to the branch

**Review by theomoondev — APPROVED** [review]
Yes, excellent! The only suggestions I have are related to drying up the code a little, but not necessary! Also thank you for your tips on the model method and that last edge case -- I changed mine to follow this design. :)

File: app/models/rental.rb (hunk 4 above)
Comment (dnguye2) [reply app/models/rental.rb:8]: An instance method may work, let's test it out during our zoom call~

File: app/controllers/rentals_controller.rb (hunk 3 above)
Comment (dnguye2) [reply app/controllers/rentals_controller.rb]: Nope we do not, accidentally left it there and forgot to take it out
